Packs/FalconX/Integrations/FalconX/FalconX.py

```diff
@@ -0,0 +1,415 @@
+from typing import Dict, Callable, Tuple, Any
+
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+import json
+import requests
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+URL = "https://api.crowdstrike.com/"
+
+
+class Client(BaseClient):
+    """
+    Client to use in the FalconX integration. Overrides BaseClient
+    """
+
+    def __init__(self, server_url: str, username: str, password: str):
+        self._base_url = server_url
+        self._username = username
+        self._password = password
+        self.token = self._generate_token()
+
+    def http_request(self, method, url_suffix, data=None, headers=None, params=None, json=None, response_type: str = 'json'):
+        """
+        Generic request to FalconX
+        """
+        full_url = urljoin(self._base_url, url_suffix)
+
+        try:
+            result = requests.request(
+                method,
+                full_url,
+                verify=False,
+                params=params,
+                data=data,
+                headers=headers,
+                json=json
+            )
+            if not result.ok:
+                demisto.log(result.text)
+                raise ValueError(f'Error in API call to FalconX {result.status_code}. Reason: {result.text}')
+
+            if response_type != 'json':
+                return result.text
+            return result.json()
+
+        except Exception as exception:
+            raise Exception(str(exception))
+
+    def post_http_req_falconx_json(self, url_suffix, data):
+        headers = {
+            'Authorization': 'bearer ' + self.token,
+            'Content-Type': 'application/json',
+        }
+
+        return self.http_request("POST", url_suffix, headers=headers, json=data)
+
+    def post_http_req_falconx_octet_stream(self, url_suffix, data):
+        headers = {
+            'Authorization': 'bearer ' + self.token,
+            'Content-Type': 'application/octet-stream',
+        }
+
+        return self.http_request("POST", url_suffix, headers=headers, data=data)
+
+    def get_http_req_falconx(self, url_suffix, params=None):
+        headers = {
+            'Authorization': 'bearer ' + self.token,
+        }
+
+        return self.http_request("Get", url_suffix, headers=headers, params=params)
```

Comment:
why can not it be part of `http_request`?
if you need one more header for some of the requests, you can add it to the args you are passing.